coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -54,38 +66,41 @@ class ShardKeyRegexPlanner(dataset: Dataset,
   private def generateExec(logicalPlan: LogicalPlan, nonMetricShardKeyFilters: Seq[ColumnFilter],
                            qContext: QueryContext): Seq[ExecPlan] = {
     val queryParams = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
-    shardKeyMatcherFn(ShardKeyMatcher(nonMetricShardKeyFilters, queryParams.promQl)).map { result =>
-        val newQueryParams = queryParams.copy(promQl = result.query)
+    shardKeyMatcher(nonMetricShardKeyFilters).map { result =>
+        val newLogicalPlan = logicalPlan.replaceFilters(result)
+        val newQueryParams = queryParams.copy(promQl = LogicalPlanUtils.logicalPlanToQuery(newLogicalPlan))
```

Comment:
Please add generous documentation on why this is needed with a few examples.
5 months later, even you may not remember ;)